### PR TITLE
Add Flow checking to Ignite

### PR DESCRIPTION
**Pull request by ryanlntn**
## Please verify the following:
- [x] Everything works on iOS/Android
- [x] `ignite-base` **ava** tests pass
- [x] `fireDrill.sh` passed

## Describe your PR
Addresses #387

I skipped Sagas and anything with a `console.tron` in it for now as I couldn't figure out how to annotate it. The type annotations are pretty basic too. It's a start though.

@GantMan Please review. 😃

### CODE REVIEW

File: ignite-base/.flowconfig [hunk 1]
```diff
@@ -15,14 +15,23 @@
...
+# Ignore failing node_modules
+.*/node_modules/react-native/Libraries/NavigationExperimental/*
+.*/node_modules/react-native/Libraries/CustomComponents/NavigationExperimental/*
+.*/node_modules/react-native/Libraries/Experimental/SwipeableRow/*
+.*/node_modules/react-native-experimental-navigation/*
+.*/node_modules/findup/test/fixture/config.json
+.*/node_modules/findup/test/fixture/f/e/d/c/b/a/top.json
+.*/node_modules/findup/test/fixture/f/e/d/c/config.json
+.*/node_modules/react-static-container/lib/StaticContainer.react.js
+.*/node_modules/react-native/Libraries/ReactIOS/YellowBox.js
```
Comment (GantMan) [inline ignite-base/.flowconfig]: hey bud, why not kill all node_modules?

File: ignite-base/.flowconfig [hunk 2]
```diff
@@ -55,4 +64,4 @@ suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
 unsafe.enable_getters_and_setters=true
 
 [version]
-^0.32.0
+^0.33.0
```
Comment (GantMan) [inline ignite-base/.flowconfig]: This errors for me: I'm seeing this issue.

https://github.com/facebook/flow/issues/2548

**Comment by GantMan** [comment]
image

File: ignite-base/.flowconfig (hunk 1 above)
Comment (ryanlntn) [inline ignite-base/.flowconfig]: If we kill all node_modules then we have to fix all the module missing errors by including them in libs. It's actually less of a PITA this way.

**Comment by ryanlntn** [comment]
image

**Comment by GantMan** [comment]
looks great
image

**Pull request merged**